### PR TITLE
feat: support kubelet credentialprovider config

### PR DESCRIPTION
**Pull request by frezbo**
Support configuring kubelet credential provider config.

Partially fixes: #7911

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -568,7 +568,7 @@ COPY ./hack/cleanup.sh /toolchain/bin/cleanup.sh
 RUN <<END
     cleanup.sh /rootfs
     mkdir -pv /rootfs/{boot/EFI,etc/cri/conf.d/hosts,lib/firmware,usr/local/share,usr/share/zoneinfo/Etc,mnt,system,opt,.extra}
-    mkdir -pv /rootfs/{etc/kubernetes/manifests,etc/cni/net.d,usr/libexec/kubernetes}
+    mkdir -pv /rootfs/{etc/kubernetes/manifests,etc/cni/net.d,usr/libexec/kubernetes,/usr/local/lib/kubelet/credentialproviders}
```
Comment (smira) [inline Dockerfile:571]: same for `arm64`?

File: internal/app/machined/pkg/controllers/k8s/kubelet_spec.go [hunk 2]
```diff
@@ -166,6 +166,12 @@ func (ctrl *KubeletSpecController) Run(ctx context.Context, r controller.Runtime
 			return fmt.Errorf("error merging arguments: %w", err)
 		}
 
+		// these flags are present till v1.24
```
Comment (smira) [inline internal/app/machined/pkg/controllers/k8s/kubelet_spec.go:169]: from 1.24?

File: Dockerfile (hunk 1 above)
Comment (frezbo) [reply Dockerfile:571]: yes, missed it

File: internal/app/machined/pkg/controllers/k8s/kubelet_spec.go (hunk 2 above)
Comment (frezbo) [reply internal/app/machined/pkg/controllers/k8s/kubelet_spec.go:169]: ah yes

**Review by smira — APPROVED** [review]

**Pull request merged**